control/controldisplay/group_counter_graph.go

```diff
@@ -14,18 +14,22 @@ type CounterGraphRenderer struct {
 	maxTotalControls int
 	segmentSize      float64
 
-	failedColorFunc colorFunc
+	options CounterGraphRendererOptions
 }
 
-func NewCounterGraphRenderer(failedControls, totalControls, maxTotalControls int, colorFuncForFailed colorFunc) *CounterGraphRenderer {
+type CounterGraphRendererOptions struct {
+	FailedColorFunc colorFunc
+}
+
+func NewCounterGraphRenderer(failedControls, totalControls, maxTotalControls int, options CounterGraphRendererOptions) *CounterGraphRenderer {
 	renderer := &CounterGraphRenderer{
 		failedControls:   failedControls,
 		totalControls:    totalControls,
 		maxTotalControls: maxTotalControls,
 		// there are 10 segments - determine the value of each segment
 		segmentSize: float64(maxTotalControls) / 10.0,
 
-		failedColorFunc: colorFuncForFailed,
+		options: options,
```

Comment:
tbh I was thinking of the options more as a way of  passing the data into the function in an extensible way - I think we're fine to keep the `FailedColorFunc` property on the renderer.

What do you think?